editor/src/components/canvas/grid-panel.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react'
+import { NO_OP } from '../../core/shared/utils'
+import { UtopiaTheme, colorTheme } from '../../uuiui'
+import { LeftPanelMinWidth } from '../editor/store/editor-state'
+import { LeftPaneComponent } from '../navigator/left-pane'
+import { CodeEditorPane, ResizableRightPane } from './design-panel-root'
+import { useGridPanelDragInfo, useGridPanelDropArea } from './grid-panels-dnd'
+import type { GridPanelData, LayoutUpdate, StoredPanel } from './grid-panels-state'
+import { GridPanelVerticalGapHalf, GridPanelsNumberOfRows } from './grid-panels-state'
+
+interface GridPanelProps {
+  onDrop: (itemToMove: StoredPanel, newPosition: LayoutUpdate) => void
+  canDrop: (itemToMove: StoredPanel, newPosition: LayoutUpdate) => boolean
+  pane: GridPanelData
+}
+
+export const GridPanel = React.memo<GridPanelProps>((props) => {
+  const { onDrop, canDrop } = props
+  const { panel, index, span, order } = props.pane
+
+  const { isDragActive, draggedPanel } = useGridPanelDragInfo()
+
+  const dropAboveElement: LayoutUpdate = React.useMemo(
+    () => ({
+      type: 'before-index',
+      indexInColumn: order,
+      columnIndex: index,
+    }),
+    [order, index],
+  )
+
+  const dropBelowElement: LayoutUpdate = React.useMemo(
+    () => ({
+      type: 'after-index',
+      indexInColumn: order,
+      columnIndex: index,
+    }),
+    [order, index],
+  )
+
+  const canDropAbove = draggedPanel != null && canDrop(draggedPanel, dropAboveElement)
```

Comment:
Do these not need to be `useMemo` wrapped?

Reply:
not because `canDropAbove` is a boolean so there is no object reference equality to save here